presqt/api_v1/serializers/resource.py

```diff
@@ -11,16 +14,30 @@ class ResourcesSerializer(serializers.Serializer):
     id = serializers.CharField(max_length=256)
     container = serializers.CharField(max_length=256)
     title = serializers.CharField(max_length=256)
-    detail = serializers.SerializerMethodField()
+    links = serializers.SerializerMethodField()
+
+    def get_links(self, instance):
+        """
+        Translate the `links` property to a custom array of Hyperlink values.
+
+        Parameters
+        ----------
+        instance : Target Obj instance
 
-    def get_detail(self, instance):
-        reversed_url = reverse(
-            viewname='resource',
-            kwargs={
-                'target_name': self.context.get('target_name'),
-                'resource_id': instance['id']})
+        Returns
+        -------
+        Hyperlink url for Target detail API endpoint
+        """
+        list_of_actions = action_checker(self.context.get('target_name'))
+        # Build a list of endpoint_actions and compare with list of actions
+        endpoint_actions = ['resource_detail', 'resource_download', 'resource_upload']
+        resources_actions = []
+        for action in endpoint_actions:
+            if action in list_of_actions:
+                resources_actions.append(action)
```

Comment:
List comprehension, baby! Cut this down to 1 or 2 lines. This has some good examples of accomplishing an intersection https://www.geeksforgeeks.org/python-intersection-two-lists/

Reply:
Now that I think about it...I think I've already checked for an intersection at some point so maybe we should build a utility function for this! :)